src/daq_config_server/client.py

```diff
@@ -1,55 +1,44 @@
-import json
-from http.client import HTTPConnection, HTTPSConnection
 from logging import Logger, getLogger
-from typing import TypeVar
+from typing import Any, TypeVar
 
-from urllib3.util import parse_url
+import requests
 
 from .constants import ENDPOINTS
 
 T = TypeVar("T")
+BlParamDType = str | int | float | bool
 
 
 class ConfigServer:
     def __init__(self, url: str, log: Logger | None = None) -> None:
-        self._url = parse_url(url)
-        self._uri_prefix = self._url.request_uri if self._url.request_uri != "/" else ""
-        if self._url.scheme != "http" and self._url.scheme != "https":
-            raise ValueError("ConfigServer must use HTTP or HTTPS!")
-        self._Conn = HTTPSConnection if self._url.scheme == "https" else HTTPConnection
+        self._url = url.rstrip("/")
         self._log = log if log else getLogger("daq_config_server.client")
 
     def _get(
         self,
         endpoint: str,
         item: str | None = None,
         options: dict[str, str] | None = None,
+        data: dict[str, Any] | None = None,
     ):
-        req_item = f"/{item}" if item else ""
-        conn = self._Conn(self._url.host, self._url.port or self._url.scheme)
-        req_ops = (
-            f"?{''.join(f'{k}={v}&' for k,v in options.items())}"[:-1]
-            if options
-            else ""
+        r = requests.get(
+            self._url + endpoint + (f"/{item}" if item else ""), options, json=data
```

Comment:
`json=data`? Are we sending json in GET request body? Seems like a bad idea. What is this really doing?

Reply:
yes, for getting a specific list of beamline params. what do you think is a better way of doing it? query params don't nicely support putting a list in there